be2-scala/src/main/scala/ch/epfl/pop/pubsub/graph/validators/ElectionValidator.scala

```diff
@@ -57,7 +72,11 @@ object ElectionValidator extends MessageDataContentValidator with EventValidator
         }*/
         // FIXME: check the actual votes
         // FIXME: for the VoteElection list, we need to check question ids but what do they mean? No info in documentation
-        else {
+        else if (!validateAttendee(sender, channel)){
+          Right(validationError(s"invalid PoP token"))
+        } else if (!validateChannelType(ObjectType.ELECTION, channel)) {
```

Comment:
The thing with this implementation is that a message may be invalid for different reasons, but the validator only signals an error for the first failing check. We may have to consider another approach to detect and report all errors rather than using cascading IF-ELSE statements. This remark also applies for other validation functions.

Reply:
I agree, however it seems like a pretty big task in such a short time to me